app/products/calls/components/current_call_bar/current_call_bar.tsx

```diff
@@ -156,42 +189,82 @@ const CurrentCallBar = ({
-                    <Text style={style.currentChannel}>{`~${displayName}`}</Text>
-                </View>
-                <Pressable
-                    onPressIn={goToCallScreen}
-                    style={style.pressable}
-                >
-                    <CompassIcon
-                        name='arrow-expand'
-                        size={24}
-                        style={style.expandIcon}
-                    />
-                </Pressable>
-                <TouchableOpacity
-                    onPress={muteUnmute}
-                    style={style.pressable}
-                >
-                    <CompassIcon
-                        name={myParticipant?.muted ? 'microphone-off' : 'microphone'}
-                        size={24}
-                        style={[style.micIcon, myParticipant?.muted ? style.muted : undefined]}
+        <>
+            <View style={style.wrapper}>
+                <View style={style.container}>
+                    <CallAvatar
+                        userModel={userModelsDict[speaker || '']}
+                        volume={speaker ? 0.5 : 0}
+                        serverUrl={currentCall?.serverUrl || ''}
                     />
-                </TouchableOpacity>
+                    <View style={style.userInfo}>
+                        <Text style={style.speakingUser}>{talkingMessage}</Text>
+                        <Text style={style.currentChannel}>{`~${displayName}`}</Text>
+                    </View>
+                    <Pressable
+                        onPressIn={goToCallScreen}
+                        style={style.pressable}
+                    >
+                        <CompassIcon
+                            name='arrow-expand'
+                            size={24}
+                            style={style.expandIcon}
+                        />
+                    </Pressable>
+                    <TouchableOpacity
+                        onPress={muteUnmute}
+                        style={[style.pressable, style.micIconContainer, myParticipant?.muted && style.muted]}
+                        disabled={!micPermissionsGranted}
+                    >
+                        <UnavailableIconWrapper
+                            name={myParticipant?.muted ? 'microphone-off' : 'microphone'}
+                            size={24}
+                            unavailable={!micPermissionsGranted}
+                            style={[style.micIcon]}
+                        />
+                    </TouchableOpacity>
+                </View>
             </View>
-        </View>
+            {micPermissionsError &&
+                <View style={style.errorWrapper}>
+                    <Pressable onPress={goToSettings}>
+                        <View style={style.errorBar}>
+                            <CompassIcon
+                                name='microphone-off'
+                                style={[style.errorIcon, {paddingRight: 9}]}
```

Comment:
good point, done -- couldn't think of a way to combine it because I'm reusing the style in one other place, so just made a `paddingRight` style. At least it avoids the new object.